library/tinkerpop/src/main/java/uk/gov/gchq/gaffer/tinkerpop/GafferPopGraph.java

```diff
@@ -980,18 +981,40 @@ private List<ElementSeed> getElementSeeds(final Iterable<Object> ids) {
                         .split(","));
             // Assume entity ID as fallback
             } else {
-                seeds.add(new EntitySeed(id));
+                seeds.add(new EntitySeed(getIDAsRelevantType(id)));
             }
 
             // If found a list verify source and destination
             if (edgeIdList.size() == 2) {
-                seeds.add(new EdgeSeed(edgeIdList.get(0), edgeIdList.get(1)));
+                Object source = getIDAsRelevantType(edgeIdList.get(0));
+                Object dest = getIDAsRelevantType(edgeIdList.get(1));
+                seeds.add(new EdgeSeed(source, dest));
             }
         });
 
         return seeds;
     }
 
+    /**
+     * Returns a the relevant Object e.g. {@link TypeSubTypeValue} from the
+     * supplied ID. As a fallback will give back the original ID if no
+     * relevant type was found.
+     *
+     * @param id The ID.
+     * @return The ID as its relevant type.
+     */
+    private Object getIDAsRelevantType(final Object id) {
+        if (id instanceof String) {
+            // See if can split into a TSTV
+            String[] split = ((String) id).split("\\|");
+            if (split.length == 3) {
+                LOGGER.debug("Parsing ID as a TSTV: {}", id);
+                return new TypeSubTypeValue(split[0], split[1], split[2]);
```

Comment:
May be worth changing to regex to bring it in line style-wise with #3230 
But this is still very readable so happy either way

Reply:
assuming we don't have to handle any whitespace here?

Reply:
Updated to use a regex instead with a slightly tweaked format. I don't think we really want to handle white space as we can't know if that is meant to be part of the type, subtype or value so safer to take whats between the `|` delimiter as literal.